File: crates/domain-scan-core/tests/fixtures/java/classes.java
package com.example;

public class UserService {
    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public User getUser(String id) {
        return repository.findById(id);
    }

    public void createUser(String name, String email) {
        User user = new User(name, email);
        repository.save(user);
    }

    private void validateEmail(String email) {
        // validation logic
    }
}

public abstract class BaseEntity {
    protected String id;
    protected String createdAt;

    public abstract String getType();
}

class InternalHelper {
    static void doSomething() {}
}

public class Config<T> {
    private T value;

    public T getValue() {
        return value;
    }
}

File: crates/domain-scan-core/tests/fixtures/kotlin/classes.kt
package com.example

class UserService(private val repository: UserRepository) {
    fun getUser(id: String): User? {
        return repository.findById(id)
    }

    fun createUser(name: String, email: String): User {
        val user = User(name, email)
        repository.save(user)
        return user
    }

    private fun validateEmail(email: String): Boolean {
        return email.contains("@")
    }
}

abstract class BaseEntity {
    abstract val id: String
    abstract fun getType(): String
}

open class Config<T>(val value: T) {
    fun getValue(): T = value
}

internal class InternalHelper {
    companion object {
        fun doSomething() {}
    }
}

object Singleton {
    fun instance(): Singleton = this
}
